src/request_builder.rs

```diff
@@ -224,13 +224,11 @@ impl RequestBuilder {
     }
 
     /// Create a `Client` and send the constructed `Request` from it.
-    pub fn send(self) -> BoxFuture<'static, Result<Response>> {
-        if let Some(client) = self.client {
-            client.send(self.req.unwrap()) // Ownership nonsense, same as `build()`.
-        } else {
-            let client = Client::new_shared();
-            client.send(self.build())
-        }
+    pub fn send(mut self) -> BoxFuture<'static, Result<Response>> {
+        self.client
+            .take()
+            .unwrap_or_else(Client::new_shared_or_panic)
+            .send(self.build())
```

Comment:
Oh, this is nicer!

Reply:
If this code didn't exist this could probably point to `Client::default()` which could use a shared client if available, changing this to `.unwrap_or_default()`, and eliminating `new_shared()` in the process? Maybe that would be confusing as it would differ from `new()` though.